pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -37,13 +38,29 @@ def _target_satellite_host(request, satellite_factory):
         yield
 
 
+@lru_cache
+def cached_capsule_cdn_register(hostname=None):
+    cap = Capsule.get_host_by_hostname(hostname=hostname)
+    cap.enable_capsule_downstream_repos()
+
+
 @contextmanager
 def _target_capsule_host(request, capsule_factory):
-    if 'sanity' not in request.config.option.markexpr:
+    if 'sanity' not in request.config.option.markexpr and not request.config.option.n_minus:
         new_cap = capsule_factory()
         yield new_cap
         new_cap.teardown()
         Broker(hosts=[new_cap]).checkin()
+    elif request.config.option.n_minus:
+        if not settings.capsule.hostname:
+            hosts = Capsule.get_hosts_from_inventory(filter="'cap' in @inv.name")
+            settings.capsule.hostname = hosts[0].hostname
+            cap = hosts[0]
+        else:
+            cap = Capsule.get_host_by_hostname(settings.capsule.hostname)
+        # Capsule needs RHEL contents for some tests
+        cached_capsule_cdn_register(hostname=settings.capsule.hostname)
```

Comment:
why capsule registration to CDN and repo enable required to be cached here?

Reply:
Since we are using single capsule, and feeded from CI for n-minus testing, we dont need to register it again and again.